File: MobiFon.Core/Entities/Identity/ApplicationUserRole.cs
using Microsoft.AspNetCore.Identity;
using MobiFon.Core.Entities.Base;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MobiFon.Core.Entities.Identity
{
    public class ApplicationUserRole: IdentityUserRole<int>, IBaseEntity
    {
        public int Id { get; set; }
        public ApplicationUser User { get; set; }
        public ApplicationRole Role { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? ModifiedAt { get; set; }
        public bool IsDeleted { get; set; }
    }
}

File: UI/propertease_client/lib/models/application_user_role.dart
import 'package:json_annotation/json_annotation.dart';

import 'application_role.dart';
import 'application_user.dart';

part 'application_user_role.g.dart';

@JsonSerializable()
class ApplicationUserRole {
  int? id;
  ApplicationUser? user; // Use the ApplicationUser class
  ApplicationRole? role; // Use the ApplicationRole class
  DateTime? createdAt;
  DateTime? modifiedAt;
  bool? isDeleted;

  ApplicationUserRole({
    this.id,
    this.user,
    this.role,
    this.createdAt,
    this.modifiedAt,
    this.isDeleted,
  });

  factory ApplicationUserRole.fromJson(Map<String, dynamic> json) =>
      _$ApplicationUserRoleFromJson(json);

  Map<String, dynamic> toJson() => _$ApplicationUserRoleToJson(this);
}
